src/libraries/Microsoft.PowerFx.Core/Public/Values/DateTimeValue.cs

```diff
@@ -15,6 +15,46 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public class DateTimeValue : PrimitiveValue<DateTime>
     {
+        /// <summary>
+        /// Direct use of <see cref="Value"/> is prohibited in favor of <see cref="GetConvertedValue(TimeZoneInfo)"/> method.
+        /// </summary>
+        public new DateTime Value => throw new NotSupportedException(string.Format("Use Method {0} instead, with proper timezone information.", nameof(GetConvertedValue)));
```

Comment:
>public new DateTime Value => throw new NotSupportedException(string.Format("Use Method {0} instead, with proper timezone information.", nameof(GetConvertedValue))); [](http://example.com/codeflow?start=8&length=164)

Note: This is a breaking change.

Reply:
We could use [Obsolete] attribute on this and still make it work. 
[Obsolete(...)]
public new DateTime Value => base.Value
 
We'll get a warning in our own code, which is probably sufficient. .
But then determined callers can still access the Value they passed in.

Reply:
done.